datalake/src/main/java/com/sequenceiq/datalake/controller/sdx/SdxInternalController.java

```diff
@@ -0,0 +1,32 @@
+package com.sequenceiq.datalake.controller.sdx;
+
+import javax.inject.Inject;
+import javax.validation.Valid;
+
+import org.springframework.stereotype.Controller;
+
+import com.sequenceiq.cloudbreak.auth.ThreadBasedUserCrnProvider;
+import com.sequenceiq.datalake.api.endpoint.sdx.SdxClusterResponse;
+import com.sequenceiq.datalake.api.endpoint.sdx.SdxInternalClusterRequest;
+import com.sequenceiq.datalake.api.endpoint.sdx.SdxInternalEndpoint;
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.service.sdx.SdxService;
+
+@Controller
+public class SdxInternalController implements SdxInternalEndpoint {
+
+    @Inject
+    private ThreadBasedUserCrnProvider threadBasedUserCrnProvider;
```

Comment:
Please note that this class has been renamed in https://github.com/hortonworks/cloudbreak/commit/4c995e6bbe9b3dff2fed43a825309cf3ad5bb23c#diff-7d7046d951f38d21f833e06689240cfe to correct for the typo (`Base` vs `Based`) and also to eliminate duplications in the Environment Service.

Reply:
Yes, thanks, I realized it it when the compilation failed :D